README.md

```diff
@@ -1,3 +1,151 @@
+    └── src               # Frontend source code
+        ├── app           # Angular JS 17 app & components
+        ├── assets        # Static assets
+        ├── environments  # Environment configuration
+        └── lib           # API, types & utilities
+```
+
+## Firebase
+
+This project uses Firebase as its backend. The configuration can be found in the [`.firebaserc`](./.firebaserc) and [`firebase.json`](./firebase.json) files.
+
+Install the firebase cli tools with the following commands:
+
+```bash
+npm install -g firebase-tools
+firebase login  # Login to the Google account destined to manage the Firebase project
+```
+
+### Emulators
+
+In order to run offline and for development purposes, we use Java Firebase emulators.
+
+```bash
+export JAVA_TOOL_OPTIONS="-Xmx4g"  # Set your desired max RAM (here: 4GB)
+firebase emulators:start  # Start the emulators
+```
+
+You will then be able to access the following UIs:
+
+- Authentication UI: `http://localhost:4000/auth`
+- Emulator UI: `http://localhost:4000`
+- Firestore UI: `http://localhost:4000/firestore`
+- Cloud Functions UI: `http://localhost:4000/functions`
+
+### Firestore
+
+We use Firestore as our database. The rules are located in the [`firestore.rules`](./firestore.rules) file.
+Basically, the database cannot be externally accessed, and must be interacted with through cloud functions.
+
+A database prototype schema can be found [here on dbdiagrams.io](https://dbdiagram.io/d/Firebase-LLM-Mediation-660d473a03593b6b61123f24) (readonly, change it with your own if you take-on the project).
+
+You can seed the database with the default data by running the following command:
+
+```bash
+curl "http://127.0.0.1:5001/friendlychat-d6dc5/us-central1/seedDatabase?seeder_password=seeder_password"
+```
+
+This calls a cloud functions that adds to the database the default data. You may want to clear it first.
+
+### Cloud Functions
+
+We use Firebase Cloud Functions to run server-side code. The functions are located in the [`functions`](./functions) directory.
+
+```bash
+cd functions
+cp .env.example .env  # Copy the example environment file
+npm install  # Install the dependencies
+npm run build:watch # Build the functions and watch for file changes for rebuilding
+```
+
+Upon running `npm run build`, if the emulator is running, it will automatically reload the functions.
+
+### Authentication
+
+This project sets up Firebase Authentication with email/password and Google sign-in.
+
+#### Email - Password authentication
+
+By default, experiment participants log in using their `uid`. It is passed to the Firebase Authentication service as `email:{uid}@palabrate` and `password:{uid}`.
+
+#### Google sign-in
+
+The Google sign-in is used for the administrators. They can log in using their Google account.
+
+In order to create a dummy google account when running the project locally with emulators, follow these steps:
+
+1. Launch the frontend and click on the Google sign-in button.
```

Comment:
I would recommend we make a TODO to make this into a small typescript script that does this for us so we don't have to depend on UI, and we can later better automate local testing/management

Reply:
I have not found a way to create a fake user with Google.com as an authentication provider programmatically yet, I created an issue for this: #9